### PR TITLE
Comynssj code review2

### PR DESCRIPTION
**Pull request by comynssj**
For changes in this code review, see comynssj_codereview branch for my most recent additions. I found this code to be both educational and easy to read. In my analysis I deduce that this program is coming along well as it was compliable when I went to review. I understood what was happening and the pages are looking clean. I made a few changes to the program, changing some of the h1 tag information to improve clarity between the pages. I also changed the givenemployeesignon method from private to public to improve future testing. I also followed the suggestion of adding @interface before the springboot test.

I did learn some new things I will try to implement in some of my other programs.
first one being the create employee test
    @Test
    void createEmployee() {
        givenEmployeeOnSignupPage();
        whenRequiredFieldsAreSubmitted();
        thenEmployeeAccountCreated();
    }

another thing I learned was how to go about doing the payroll form, this can certainly be applied to our inventory program
    <form action="#" th:action="@{submitPayroll}" th:object="${payroll}" method="POST">
        <label>Employee</label>
        <input type="text" name="employee" th:field="*{employee}"/>
        <label>Date:</label>
        <input type="text" name="date" th:field="*{date}"/>
        <label>Hours Worked:</label>
        <input type="text" name="hours" th:field="*{hoursWorked}"/>
        <label>Amount Paid</label>
        <input type="text" name="hours" th:field="*{amountPaid}"/>
        <button name="submit" type="submit" value="0" class="btn btn-primary">Submit</button>
    </form>

Finally I learned more about using the th:each and th:text attributes
    <tr th:each="payroll: ${payrollService.log}">
        <td th:text="${payroll.employee.getFullName()}"></td>
        <td th:text="${payroll.date}"></td>
        <td th:text="${payroll.hoursWorked}"></td>
        <td th:text="${payroll.amountPaid}"></td>
    </tr>

### CODE REVIEW

**Comment by vumd-UC** [comment]
Overall, I think this looks good, but it _does_ do a lot of the same things done in the Sprint 2 Branch.

File: src/main/java/com/timekeeper/app/controllers/TimeKeeperController.java [hunk 1]
```diff
@@ -78,7 +78,7 @@ public String submitPayroll(Payroll payroll) {
 
     @GetMapping("/all-payroll")
     @ResponseBody
-    public ArrayList<Payroll> fetchAllPayroll() {
+    public List<Payroll> fetchAllPayroll() {
```
Comment (discospiff) [inline src/main/java/com/timekeeper/app/controllers/TimeKeeperController.java:81]: Good.  We usually use interface type for variables; class type for objects.

File: src/main/java/com/timekeeper/app/dao/EmployeeRepository.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.timekeeper.app.dao;
+
+import com.timekeeper.app.dto.Employee;
+import org.springframework.data.repository.CrudRepository;
+
+// need to add following maven dependencies alt+enter
+// spring-boot-starter-data-jpa - search for artifact
+// mysql-connector-java - search for artifact
```
Comment (discospiff) [inline src/main/java/com/timekeeper/app/dao/EmployeeRepository.java:8]: It looks like this is already complete (in the pom.xml), so safe to remove these instructions.

**Comment by discospiff** [comment]
There is a bit of code review + new feature work here.  The feature work appears to be the bulk of the changes.  I recommend that the project team keep this branch open, and continue to develop the feature, if it is a feature they want to keep.

**Pull request merged**